R/knitr-engine.R

```diff
@@ -730,7 +747,13 @@ eng_python_autoprint <- function(captured, options) {
 
     return("")
 
-  } else if (py_has_method(value, "_repr_markdown_")) {
+  } else {
+    return(eng_python_generic_autoprint(captured, value))
+  }
+}
+
+eng_python_generic_autoprint <- function(captured, value) {
+  if (py_has_method(value, "_repr_markdown_")) {
```

Comment:
As commented in main comment, I wonder if we should do as Quarto and 

* Output HTML first if available
* Output Markdown if available
* Then captured (equivalent to `text/plain`) 

Logic in Quarto for display is at [`displayMimeType()`](https://github.com/quarto-dev/quarto-cli/blob/9e012de66870130595870aa76a0e28ea1c848580/src/core/jupyter/display-data.ts#L45-L106) where we have a logic of ordering possibly output before checking the output values. In this logic, if there is HTML and Markdown representation, HTML is preferred currently